### ISSUE TITLE
HLS doesn't play in IE11 on Windows 7 when dash.js is present

### ISSUE THREAD

**Issue by thomaspeeters** [videojs/videojs-contrib-hls#420]
### Project info
I'm trying to set up a way to stream using as little Flash as possible. The tech order I'm using is
1. MPEG-DASH 
2. HLS
3. RTMP (with Flash)

I'm testing in Chrome 46, Firefox 41 and IE 11 on Windows 7. 
I'm not having much luck going "Flashless", so far only Chrome 46 manages MPEG-DASH and HLS without Flash. Firefox 41 and IE11 (on win7) apparently don't do MPEG-DASH and HLS only works via a Flash component.

My file is an H.264;AAC encoded .mp4, streaming from a Wowza 4.3.0 server. 

Possibly relevant video.js settings:
- `"autoplay": false`
- `"preload": "auto"`
- `"techOrder":  ["html5", "flash"]`

### The problem
IE 11 on Windows 7 doesn't play HLS when dash.js is present while other browsers do.

I'm loading the following libraries:

- video.js 5.0.0
- videojs-media-sources 2.1.0
- videojs-contrib-hls development 1.0.1-0 (commit c7d5b62)
- dash.js 1.5.1 (dash.all.js)
- videojs-contrib-dash 2.0.0
- IE 11.0.9600.18015 on Windows 7

### Result
- Screen remains black
- No errors
- Play and seek do nothing (except for player interface changes)
- Console output (consistently the same):

`SEC7118: XMLHttpRequest for https://<...>/<...>/_definst_/mp4:<...>.mp4/playlist.m3u8 required Cross Origin Resource Sharing (CORS).`
`File: video`
`SEC7118: XMLHttpRequest for https://<...>/<...>/_definst_/mp4:<...>.mp4/chunklist_w470133660.m3u8 required Cross Origin Resource Sharing (CORS).`
`File: video`
`SEC7118: XMLHttpRequest for https://<...>/<...>/_definst_/mp4:<...>.mp4/media_w470133660_0.ts required Cross Origin Resource Sharing (CORS).`
`File: video`
`SEC7118: XMLHttpRequest for https://<...>/<...>/_definst_/mp4:<...>.mp4/media_w470133660_1.ts required Cross Origin Resource Sharing (CORS).`
`File: video`
`SEC7118: XMLHttpRequest for https://<...>/<...>/_definst_/mp4:<...>.mp4/media_w470133660_2.ts required Cross Origin Resource Sharing (CORS).`
`File: video`
`SEC7118: XMLHttpRequest for https://<...>/<...>/_definst_/mp4:<...>.mp4/media_w470133660_3.ts required Cross Origin Resource Sharing (CORS).`
`File: video`
`SEC7118: XMLHttpRequest for https://<...>/<...>/_definst_/mp4:<...>.mp4/media_w470133660_4.ts required Cross Origin Resource Sharing (CORS).`
`File: video`

### Possible cause
Leaving out the dash.js library seems to fix the problem, but this is not an acceptable solution as MPEG-DASH is my primary target.

### Result without dash.js
- Same as Firefox and Chrome. Video.js uses the video.js Flash component to play the HLS stream.
- Console output (consistently the same):

`SEC7118: XMLHttpRequest for https://<...>/<...>/_definst_/mp4:<...>.mp4/playlist.m3u8 required Cross Origin Resource Sharing (CORS).`
`File: video`
`SEC7118: XMLHttpRequest for https://<...>/<...>/_definst_/mp4:<...>.mp4/chunklist_w432919051.m3u8 required Cross Origin Resource Sharing (CORS).`
`File: video`
`SEC7118: XMLHttpRequest for https://<...>/<...>/_definst_/mp4:<...>.mp4/media_w432919051_0.ts required Cross Origin Resource Sharing (CORS).`
`File: video`
`SEC7118: XMLHttpRequest for https://<...>/<...>/_definst_/mp4:<...>.mp4/media_w432919051_1.ts required Cross Origin Resource Sharing (CORS).`
`File: video`
`SEC7118: XMLHttpRequest for https://<...>/<...>/_definst_/mp4:<...>.mp4/media_w432919051_2.ts required Cross Origin Resource Sharing (CORS).`
`File: video`

**Comment by thomaspeeters** [comment]
I've narrowed it down to the following line (using `dash.debug.js` instead of `dash.debug.js`):

```javascript
// dash.debug.js v1.5.1 line 1251
if (undefined === btoa) {
    var btoa = BASE64.encode;
}
```

Looks like dash.js implements its own base64 encoding into a globar var which conflicts with `window.btoa` in IE11.
I'm not sure why dash.js does this, especially overriding the default implementation. Is something wrong with the default browser implementation? Does anyone know more? Why does this only fail in IE11?

**Comment by dmlap** [comment]
Do you think this is an issue in dash.js then?

**Comment by thomaspeeters** [comment]
I believe it is. I forgot to mention here that I also opened an issue in the dash.js project (see: Dash-Industry-Forum/dash.js/issues/866) to address this. 
At the time of writing there has been no response there and I don't know whether or not the `btoa` and `atob` declarations are safe to remove, though I suspect they are. There is still the possibility that additional, optional components I did not compile into my dash.js file make use of the declarations. An expert over at the dash.js project should be able to answer that question.
